src/poller_base.cpp

```diff
@@ -55,7 +55,11 @@ void zmq::poller_base_t::add_timer (int timeout_, i_poll_events *sink_, int id_)
 {
     uint64_t expiration = _clock.now_ms () + timeout_;
     timer_info_t info = {sink_, id_};
+#if __cplusplus >= 201103L || (defined _MSC_VER && _MSC_VER >= 1700)
+    _timers.emplace (expiration, info);
```

Comment:
If touching this, it should probably use `ZMQ_MAP_INSERT_OR_EMPLACE`.

IIUC, this is not related to the other changes in this commit, right?

Reply:
Hi. Sorry for the late reply. I saw this construct elsewhere in the code. I wasn't aware of `ZMQ_MAP_INSERT_OR_EMPLACE`. 
I see it's defined only in `blob.hpp`. Including this file might have side effects I'm not aware of.
Do you think it should be included?

Yes, this is not related to the actual issue.

Reply:
`ZMQ_MAP_INSERT_OR_EMPLACE` should probably be moved to a different header. Maybe macros.hpp?

Can you remove this change from this commit to avoid confusion, given that the other changes are quite critical? Then we can improve this separately.

Reply:
Yes, I'll remove it. Doesn't functionally change anything.

Reply:
Done.